sdk/ai/azure-ai-assistants/azure/ai/assistants/_operations/_patch.py

```diff
@@ -935,6 +935,509 @@ def upload_file(
+         to use only the number of completion tokens specified, across multiple turns of the run. If
+         the run exceeds the number of
+         completion tokens specified, the run will end with status ``incomplete``. See
+         ``incomplete_details`` for more info. Default value is None.
+        :paramtype max_completion_tokens: int
+        :keyword truncation_strategy: The strategy to use for dropping messages as the context windows
+         moves forward. Default value is None.
+        :paramtype truncation_strategy: ~azure.ai.assistants.models.TruncationObject
+        :keyword tool_choice: Controls whether or not and which tool is called by the model. Is one of
+         the following types: str, Union[str, "_models.AssistantsApiToolChoiceOptionMode"],
+         AssistantsNamedToolChoice Default value is None.
+        :paramtype tool_choice: str or str or
+         ~azure.ai.assistants.models.AssistantsApiToolChoiceOptionMode or
+         ~azure.ai.assistants.models.AssistantsNamedToolChoice
+        :keyword response_format: Specifies the format that the model must output. Is one of the
+         following types: str, Union[str, "_models.AssistantsApiResponseFormatMode"],
+         AssistantsApiResponseFormat Default value is None.
+        :paramtype response_format: str or str or
+         ~azure.ai.assistants.models.AssistantsApiResponseFormatMode or
+         ~azure.ai.assistants.models.AssistantsApiResponseFormat
+        :keyword metadata: A set of up to 16 key/value pairs that can be attached to an object, used
+         for storing additional information about that object in a structured format. Keys may be up to
+         64 characters in length and values may be up to 512 characters in length. Default value is
+         None.
+        :paramtype metadata: dict[str, str]
+        :keyword interval: Time to wait before polling for the status of the run. Default value
+         is 1.
+        :paramtype interval: float
+        :return: ThreadRun. The ThreadRun is compatible with MutableMapping
+        :rtype: ~azure.ai.assistants.models.ThreadRun
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        run = self.create_run(
+            thread_id,
+            body=body,
+            assistant_id=assistant_id,
+            model=model,
+            instructions=instructions,
+            additional_instructions=additional_instructions,
+            additional_messages=additional_messages,
+            tools=tools,
+            temperature=temperature,
+            top_p=top_p,
+            max_prompt_tokens=max_prompt_tokens,
+            max_completion_tokens=max_completion_tokens,
+            truncation_strategy=truncation_strategy,
+            tool_choice=tool_choice,
+            response_format=response_format,
+            metadata=metadata,
+            **kwargs
+        )
+        while run.status in ["queued", "in_progress", "requires_action"]:
```

Comment:
I on-purposely wait and sleep during require_action.   It is true that if the action is never been taken, it will be poll indefinitely.   However, if the action is taken within the same process or thread, developer should not call this function to call.   If they do, they should take actions on other threads or micro-services.